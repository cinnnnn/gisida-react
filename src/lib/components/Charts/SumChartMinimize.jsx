import React from 'react';
import PropTypes from 'prop-types';

class SumChartMinimize extends React.Component {
  constructor(props) {
    super(props);
    this.handleClick = this.handleClick.bind(this);
    this.state = {
      isMin: false,
      bottom: this.props.bottom,
      label: this.props.label,
    };
  }

  componentWillReceiveProps(nextProps) {
    this.setState({ bottom: nextProps.bottom });
  }

  handleClick(e) {
    e.preventDefault();
    this.props.toggleChart();
    this.setState({ isMin: !this.state.isMin });
  }

  render() {
    return (
      <a
        className="toggleChart"
        role="button"
        tabIndex="-1"
        onClick={(e) => { this.handleClick(e); }}
        title={`${this.state.isMin ? 'Show' : 'Hide'} Summary Charts`}
        style={{ bottom: this.state.bottom }}
        data-icon-credit="Created by Barracuda from the Noun Project"
        data-icon-credit-url="https://thenounproject.com/barracuda/collection/chart/?i=1217547"
      >
        <span />
      </a>
    );
  }
}

SumChartMinimize.propTypes = {
  toggleChart: PropTypes.func.isRequired,
  label: PropTypes.string.isRequired,
  bottom: PropTypes.number.isRequired,
};

export default SumChartMinimize;
