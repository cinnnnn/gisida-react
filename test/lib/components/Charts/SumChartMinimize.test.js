import React from 'react';
import { SumChartMinimize }  from '../../../../src/lib/components/Charts/SumChartMinimize.jsx'
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';

const toggleChart = jest.fn();

const componentWrapper = shallow(
	<SumChartMinimize 
		toggleChart={toggleChart}
		label=''
		bottom={52}
	/>
);

describe('SumChartMinimize', () => {
	it('SumChartMinimize component renders correctly', () => {
		const json = toJson(componentWrapper)
		expect(json).toMatchSnapshot();
	})
});
