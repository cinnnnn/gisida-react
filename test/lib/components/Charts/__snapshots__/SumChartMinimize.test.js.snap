// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SumChartMinimize SumChartMinimize component renders correctly 1`] = `
<a
  className="toggleChart"
  data-icon-credit="Created by Barracuda from the Noun Project"
  data-icon-credit-url="https://thenounproject.com/barracuda/collection/chart/?i=1217547"
  onClick={[Function]}
  role="button"
  style={
    Object {
      "bottom": 52,
      "right": "35px",
      "zIndex": 5000,
    }
  }
  tabIndex="-1"
  title="Hide Summary Charts"
>
  <span />
</a>
`;
